src/NuGet.Core/NuGet.Repositories/NuGetv3LocalRepository.cs

```diff
@@ -69,7 +84,17 @@ public IEnumerable<LocalPackageInfo> FindPackagesById(string packageId)
                             id = Path.GetFileNameWithoutExtension(manifestFileName);
                         }
 
-                        packages.Add(new LocalPackageInfo(id, version, fullVersionDir));
+                        var sha512Path = _pathResolver.GetHashPath(id, version);
```

Comment:
nit: could we leave this named `hashPath`. That hash algorithm seems like an implementation detail.